opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -104,13 +104,43 @@ message MetricDescriptor {
   repeated string label_keys = 5;
 }
 
+// LabelValues is a list of label string values.
+message LabelValues {
+  // Values of labels. Each element in this array is the value of the key defined at the
+  // corresponding index of MetricDescriptor.label_keys array. The number of elements
+  // in these 2 arrays must be the same. Note that values array must include empty
+  // strings and must also contain an empty string value for labels for which the
+  // value was originally unspecified.
+  repeated string values = 1;
```

Comment:
I'd say aggregation of metrics would be challenging as there will be a need to track a separate array of unspecified values. Is it a concern?

Looking at API - we may end up with various label set quite often as LabelSet is not pre-created when metric is registered. @jmacd do you think it will be a common case of missing labels?

Reply:
I don't think that's a problem. When exporting we will need to Go over internal metric representation and convert in-memory LabelSet to LabelValues proto message. Should not be an issue to build the message. unspecified values don't really make this materially more difficult. It will be a straightforward array iteration and appending of indexes.
Alternatively this can be done once per LabelSet and cached.

Reply:
API declares metric and then on every measurement recording you can pass whatever LabelSet you want. If aggregation logic will attempt to merge different LabelSet's - there potentially be many missing dimensions quite often. If LabelSet will not be merged and data will be aggregated separately - we do not need a missing labels functionality

Reply:
I'm not sure what it means to "merge" label sets, as nothing like that was included in the specification. There has been _speculation_ that people will demand more of label sets, such as a way to extend them (e.g., `labelSet.Add(oneMoreLabel)`, or `labelSet.Combine(more, label, sets)`).

LabelSets contain arbitrary labels and are not bound to an instrument. Since instruments may be aggregated along different dimensions, it's certainly possible to have unspecified values. I think it's prudent to preserve this information -- otherwise there will be ambiguity.

In the currently open WIP PR adding a stdout exporter in Go (https://github.com/open-telemetry/opentelemetry-go/pull/265), I left a TODO about this: https://github.com/open-telemetry/opentelemetry-go/blob/63d79a833e3312b89b2de9fcfbe744561fe70563/sdk/metric/batcher/stateful/stateful.go#L76